apps/meteor/app/api/server/v1/chat.ts

```diff
@@ -822,3 +823,30 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'chat.getURLPreview',
+	{ authRequired: true },
+	{
+		async get() {
+			const { roomId, url } = this.queryParams;
+
+			if (!roomId) {
+				throw new Meteor.Error('error-roomId-param-not-provided', 'The required "roomId" query param is missing.');
+			}
+
+			if (!url) {
+				throw new Meteor.Error('error-url-param-not-provided', 'The required "url" query param is missing.');
+			}
+
+			if (!(await canAccessRoomIdAsync(roomId, this.userId))) {
+				throw new Meteor.Error('error-not-allowed', 'Not allowed');
+			}
+
+			const { urlPreview } = await OEmbed.parseUrl(url);
+			urlPreview.ignoreParse = true;
```

Comment:
can you explain why is this needed? or what is the purpose of this?

Reply:
The idea for adding this param here is that we don't parse the same URL again when the output of this endpoint is "attached" to a message and sent to RC.
Since the final purpose of this endpoint is being used by our client to generate previews before sending a message, I thought it'd make sense to assign this param here so we don't parse this URL again later when the message is sent (I felt like this should be this endpoint's responsibility, but please tell me if you think otherwise).